src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -621,6 +625,48 @@ def get_node_os_upgrade_channel(self) -> Union[str, None]:
         # this parameter does not need validation
         return node_os_upgrade_channel
 
+    def get_upgrade_override_until(self) -> Union[str, None]:
+        """Obtain the value of upgrade_override_until.
+        :return: string or None
+        """
+        # this parameter does not need dynamic completion
+        # this parameter does not need validation
+        return self.raw_param.get("upgrade_override_until")
+
+    def get_upgrade_settings(self) -> Union[List[str], None]:
+        """Obtain the value of upgrade_settings.
+        :return: List[str] or None
+        """
+        # this parameter does not need dynamic completion
+        # this parameter does not need validation
+        upgrade_settings = self.raw_param.get("upgrade_settings")
+        if upgrade_settings is None or len(upgrade_settings.strip()) == 0:
+            return None
```

Comment:
It's a little bit weird to disable the feature by setting the option value to a string "None".

All other similar parameters (`cluster_autoscaler_profile`, `api_server_authorized_ip_ranges`, `nodepool_labels`) are using empty string to remove the previous settings. From cli perspective, in `az aks update`, it takes empty string (`""`, i.e., `--api-server-authorized-ip-ranges=""`) to to remove the previous settings and `None` (default value, not specifying this option) to keep the previous settings (backfill on rp side).

Reply:
Updated